code/game/gamemodes/special_agent/special_agent.dm

```diff
@@ -0,0 +1,78 @@
+/*!
+Basically the traitor game mode but with an agent who works against the other syndicate operatives, attempting to extract one or more of them.
+*/
+/datum/game_mode/traitor/special_agent
+	name = "traitor+specialagent"
+	config_tag = "traitoragent"
+	report_type = "traitoragent"
+	antag_flag = ROLE_SPECIAL_AGENT
+	false_report_weight = 5
+	restricted_jobs = list("AI","Cyborg")
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain")
+	required_players = 15
+	required_enemies = 1
+	recommended_enemies = 5
+	reroll_friendly = 1
+
+	announce_span = "danger"
+	announce_text = "There are Syndicate Agents and Special Agents on the Station!\n\
+	<span class='danger'>Traitors</span>: Accomplish your objectives!\n\
+	<span class='danger'>Special Agents</span>: Accomplish your objectives!\n\
+	<span class='notice'>Crew</span>: Do not let the traitors or special agents succeed!"
+
+	var/list/datum/mind/pre_agents = list()
+	var/list/datum/mind/agents = list()
+/datum/game_mode/traitor/special_agent/pre_setup()
+	if(CONFIG_GET(flag/protect_roles_from_antagonist))
+		restricted_jobs += protected_jobs
+	if(CONFIG_GET(flag/protect_assistant_from_antagonist))
+		restricted_jobs += "Assistant"
+	
+	var/list/datum/mind/possible_agents = get_players_for_role(ROLE_SPECIAL_AGENT)
+	
+	var/num_agents = 1
+	
+	var/sasc = CONFIG_GET(number/special_agent_scaling_coeff)
+	if(sasc)
+		num_agents = max(1, min(round(num_players() / (sasc * 2)) + 2 + num_modifier, round(num_players() / sasc) + num_modifier))
```

Comment:
the logic here is a little fraught, and can be made a lot clearer to someone reading by using some extra lines and vars, especially if named well.

I'd calculate the amounts first
```
amount = round(num_players() (sasc * 2) + 2 + num_modifier)
amount2 = round(num_players() / sasc) + num_modifer)
max(1, min(amount, amount2))
```
why the two different calculations?

Reply:
I just ripped the calcs from the traitor file, it confused the fuck out of me too which is why I didn't touch it.

Reply:
`var/tsc = CONFIG_GET(number/traitor_scaling_coeff)
	if(tsc)
		num_traitors = max(1, min(round(num_players() / (tsc * 2)) + 2 + num_modifier, round(num_players() / tsc) + num_modifier))
	else
		num_traitors = max(1, min(num_players(), traitors_possible))`

Reply:
I changed it but I still don't understand the math